source/backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -91,14 +93,34 @@ public PimsActivityInstance Add(PimsActivityInstance instance)
             return instance;
         }
 
-        public PimsActivityInstance Update(PimsActivityInstance instance)
+        public Tuple<PimsActivityInstance, PimsActivityInstanceNote> Update(PimsActivityInstance instance)
         {
             instance.ThrowIfNull(nameof(instance));
             var currentActivity = this.Context.PimsActivityInstances
                 .FirstOrDefault(x => x.ActivityInstanceId == instance.Id) ?? throw new KeyNotFoundException();
+
+            PimsActivityInstanceNote note = null;
+            if (currentActivity.ActivityInstanceStatusTypeCode!=instance.ActivityInstanceStatusTypeCode && this.User.HasPermission(Permissions.NoteAdd))
```

Comment:
nit: fix spacing around `!=`

Reply:
Also, come to think about - not sure about the permission check for current user. I believe it's the "system" creating a note automatically so it would be irrelevant if the user has claims for notes or not. What are your thoughts on this @devinleighsmith ?

My understanding is that we would want to log notes per status change regardless of the claims of the user who triggered this change - is it not?

Reply:
Yes, in this case as long as the user has the ability to update an activity the system should be able to generate the note, regardless of their note permissions.

Reply:
@devinleighsmith this was a precondition in the story that user needs to have a Create permission on the note.
I will remove it as per above discussion.